### DISCUSSION TITLE
Partner Requesting IP (Serverless Function)

### DISCUSSION THREAD

**Discussion by ajayvignesh01** [vercel/community#3443] [Help] [answered]
### Summary

I have a deployment on Next.js that uses vercel serverless functions.

In a nutshell, when our user clicks a button on our website, a serverless function will be triggered. In that function, a POST will be made to our partner's API. 

The issue here is that, our partner is asking for our IP address where we make the API calls from so that they can whitelist it and allow access.

How do I find out the IP address of my serverless functions?

### Steps to Reproduce

Run Serverless Function

**Comment by amyegan** [comment, marked as answer]
Hey @0x-Legend. Due to the dynamic nature of serverless functions, the IP address will not always be the same. It is recommended that you allow all IP addresses and secure your database with a randomly generated password stored as an [environment variable](https://vercel.com/docs/concepts/projects/environment-variables) at least 32 characters in length.

For more info, check out this guide: https://vercel.com/guides/how-to-allowlist-deployment-ip-address
